### PR TITLE
Set Accept header in JSON requests

### LINKED ISSUE #73: `encode_body`: Set `Accept` header on `json: body`

### PR DESCRIPTION
**Pull request by vinibrsl**
Closes #73

### CODE REVIEW

File: lib/req/steps.ex [hunk 1]
```diff
@@ -285,6 +285,7 @@ defmodule Req.Steps do
       data = request.options[:json] ->
         %{request | body: Jason.encode_to_iodata!(data)}
         |> Req.Request.put_new_header("content-type", "application/json")
+        |> Req.Request.put_new_header("accept", "application/json, */*;q=0.5")
```
Comment (wojtekmach) [inline lib/req/steps.ex:288]: does curl also set `*/*` in these circumstances? If not let's not do that yet.

Comment (wojtekmach) [inline lib/req/steps.ex:288]: please also document that we're setting the accept header. Perhaps something like this:

```markdown
* `:json` - (...) It also sets the `accept` header to `application/json`.
```

Comment (vinibrsl) [reply lib/req/steps.ex:288]: `curl` actually only sets `*/*`. I took the `application/json, */*;q=0.5` example from [httpie](https://github.com/httpie/httpie).

## curl
```
curl -v -X POST https://reqbin.com/echo/post/json -H 'Content-Type: application/json' -d '{"hello":"world"}'

> POST /echo/post/json HTTP/2
> Host: reqbin.com
> user-agent: curl/7.79.1
> accept: */*
> content-type: application/json
> content-length: 17
```

## httpie
```
http -v PUT pie.dev/put name=John email=john@example.org

Accept: application/json, */*;q=0.5
Accept-Encoding: gzip, deflate
Connection: keep-alive
Content-Length: 45
Content-Type: application/json
Host: pie.dev
User-Agent: HTTPie/3.1.0
```

Comment (wojtekmach) [reply lib/req/steps.ex:288]: sorry, I specifically meant the `curl --json` option. It sets application/json:

```
curl --json '{ "drink": "coffe" }' https://httpbin.org/anything
{
  "args": {},
  "data": "{ \"drink\": \"coffe\" }",
  "files": {},
  "form": {},
  "headers": {
    "Accept": "application/json",
    "Content-Length": "20",
    "Content-Type": "application/json",
    "Host": "httpbin.org",
    "User-Agent": "curl/7.83.0",
    "X-Amzn-Trace-Id": "Root=1-62b32ce1-34c3c2874e099e1c361c5b28"
  },
  "json": {
    "drink": "coffe"
  },
  "method": "POST",
  "origin": "91.150.170.123",
  "url": "https://httpbin.org/anything"
}
```

let's go with that but I'm happy to adjust down the road.

File: lib/req/steps.ex [hunk 2]
```diff
@@ -264,7 +264,9 @@ defmodule Req.Steps do
 
     * `:form` - if set, encodes the request body as form data (using `URI.encode_query/1`).
 
-    * `:json` - if set, encodes the request body as JSON (using `Jason.encode_to_iodata!/1`).
+    * `:json` - if set, encodes the request body as JSON (using `Jason.encode_to_iodata!/1`), sets
+                the `accept` header to `application/json, */*;q=0.5`, and the `content-type`
```
Comment (wojtekmach) [inline lib/req/steps.ex:268]:
```suggestion
                the `accept` header to `application/json`, and the `content-type`
```

File: lib/req/steps.ex [hunk 3]
```diff
@@ -285,6 +287,7 @@ defmodule Req.Steps do
       data = request.options[:json] ->
         %{request | body: Jason.encode_to_iodata!(data)}
         |> Req.Request.put_new_header("content-type", "application/json")
+        |> Req.Request.put_new_header("accept", "application/json, */*;q=0.5")
```
Comment (wojtekmach) [inline lib/req/steps.ex:290]:
```suggestion
        |> Req.Request.put_new_header("accept", "application/json")
```

File: test/req/steps_test.exs [hunk 4]
```diff
@@ -166,9 +166,16 @@ defmodule Req.StepsTest do
     assert "req/" <> _ = Req.get!(c.url).body
   end
 
-  test "encode_body/1: json" do
-    req = Req.new(json: %{a: 1}) |> Req.Request.prepare()
-    assert req.body |> IO.iodata_to_binary() == ~s|{"a":1}|
+  test "encode_body/1: json", c do
+    Bypass.expect(c.bypass, "POST", "/", fn conn ->
+      assert {:ok, ~s|{"a":1}|, conn} = Plug.Conn.read_body(conn)
+      assert ["application/json, */*;q=0.5"] = Plug.Conn.get_req_header(conn, "accept")
```
Comment (wojtekmach) [inline test/req/steps_test.exs:172]:
```suggestion
      assert ["application/json"] = Plug.Conn.get_req_header(conn, "accept")
```

**Comment by wojtekmach** [comment]
Thanks!

**Pull request merged**